src/models/Discount.php

```diff
@@ -142,11 +142,12 @@ class Discount extends Model
      * @var bool The whole order has free shipping.
      */
     public $hasFreeShippingForOrder;
-
+    
     /**
-     * @var bool Match all user groups.
+     * @var string User condition type.
      */
-    public $allGroups;
+    public $userCondition;
```

Comment:
Being picky, but maybe this would be more accurate as `$userGroupCondition` with a more explanatory description like “Type of user group condition that should match the discount. (See `getUserConditions()`.)”